### PR TITLE
Added pipeWith/composeWith functions

### PR DESCRIPTION
**Pull request by skallet**
Squashed commits from #2515.

### CODE REVIEW

**Comment by skallet** [comment]
@GingerPlusPlus Fixed, sorry I've missed it in last update. It has been so long I made any changes to this PR 💃

File: source/composeWith.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Performs right-to-left function composition using transforming function. The rightmost function may have
+ * any arity; the remaining functions must be unary.
+ *
+ * **Note:** The result of compose is not automatically curried.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig (((a, b, ..., n) -> o), [(y -> z), (x -> y), ..., (o -> p)]) -> ((a, b, ..., n) -> z)
```
Comment (Bradcomp) [inline source/composeWith.js]: This signature isn't right. Sorry I didn't catch it the first time. The first function is interpolated between all the others, so it won't take the initial parameters. For lack of a better option, I think I'd be OK with `* -> *`

**Comment by skallet** [comment]
Updated in composeWith/pipeWith fns.

**Comment by Bradcomp** [comment]
When I said `(* -> *)` I just meant for the first part of it, not the whole thing. Sorry I wasn't clearer :(

**Comment by skallet** [comment]
So this
`* @sig (((a, b, ..., n) -> o), [(y -> z), (x -> y), ..., (o -> p)]) -> ((a, b, ..., n) -> z)`
Become this?
`* @sig (((* -> *), [(y -> z), (x -> y), ..., (o -> p)]) -> ((a, b, ..., n) -> z)`

**Comment by Bradcomp** [comment]
Yes, but let’s pull the a,b,... into the array as well.

**Comment by skallet** [comment]
@Bradcomp please check update.

**Comment by Bradcomp** [comment]
I meant this:

`((* -> *), [(y -> z), (x -> y), ..., (o -> p), ((a, b, ..., n) -> o)]) -> ((a, b, ..., n) -> z)

**Comment by skallet** [comment]
@Bradcomp is it alright now?

File: source/pipeWith.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * Performs left-to-right function composition using transforming function. The leftmost function may have
+ * any arity; the remaining functions must be unary.
+ *
+ * **Note:** The result of pipeWith is not automatically curried.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig ((* -> *), [(o -> p), ..., (x -> y), (y -> z), ((a, b, ..., n) -> y)]) -> ((a, b, ..., n) -> z)
```
Comment (Bradcomp) [inline source/pipeWith.js]: For `pipeWith`, the variadic function should be the first item in the array instead of the last

**Pull request merged**